src/table/tools-header.tsx

```diff
@@ -29,10 +30,26 @@ export default function ToolsHeader({ header, filter, pagination, preferences }:
       {isHeaderString ? <span id={headingId}>{header}</span> : header}
       {hasTools && (
         <div ref={ref} className={clsx(styles.tools, isSmall && styles['tools-small'])}>
-          {filter && <div className={styles['tools-filtering']}>{filter}</div>}
+          {filter && (
+            <div
+              className={styles['tools-filtering']}
+              onClickCapture={() => setLastUserAction?.('filter')}
+              onKeyDownCapture={() => setLastUserAction?.('filter')}
+            >
+              {filter}
+            </div>
+          )}
           <div className={styles['tools-align-right']}>
-            {pagination && <div className={styles['tools-pagination']}>{pagination}</div>}
-            {preferences && <div className={styles['tools-preferences']}>{preferences}</div>}
+            {pagination && (
+              <div className={styles['tools-pagination']} onClickCapture={() => setLastUserAction?.('pagination')}>
+                {pagination}
+              </div>
+            )}
+            {preferences && (
+              <div className={styles['tools-preferences']} onClickCapture={() => setLastUserAction?.('preferences')}>
```

Comment:
Does it work only for opening preferences dialog, or any interaction inside too?

How about updating the test "emits a metric when the user changes preferences" to ensure that it captures specifically the submit button click and not relies on the whole interaction wrapping faster than 1 second?

Reply:
The suggestion to delay interactions inside preferences integration test still stands